### PR TITLE
Connect to database and return user password

### PR DESCRIPTION
**Pull request by RyanMacG**
FreeRadius expects the cleartext password of a user to be returned which
is then compared to a hashed version. This llows us to retrieve users as
identified by the FreeRadius request and return the correct password or
an empty string if they are not a real user.

### CODE REVIEW

File: spec/app_spec.rb [hunk 1]
```diff
@@ -1,13 +1,77 @@
...
+        expect(last_response.body).to eq('')
+      end
+    end
+
+    context 'with extra url parameters' do
+      let(:url) { "/authorize/user/#{username}/mac/#{client_mac}/ap/#{ap_mac}/site/#{ap_ip_address}/apg/#{ap_aruba_name}/mdn/#{ap_meraki_name}" }
+
+      context 'as a valid user' do
+        before do
+          db[:userdetails].insert(username: username, password: 'FooBarBaz')
```
Comment (adrianclay) [inline spec/app_spec.rb]: Nice.

File: spec/app_spec.rb [hunk 2]
```diff
@@ -1,13 +1,77 @@
...
+    after do
+      db[:userdetails].truncate
+    end
+
+    context 'as the Health user' do
+      let(:username) { 'HEALTH' }
+
+      before do
+        db[:userdetails].insert(username: username, password: 'TestUserPassword')
+        get "/authorize/user/#{username}"
```
Comment (adrianclay) [inline spec/app_spec.rb]: Be good to be clear under what circumstances this shorter URL gets called @afoldesi-gds and then add that context to the test description.

File: docker-compose.yml [hunk 3]
```diff
@@ -1,7 +1,23 @@
...
+      - "13306:3306"
+
   app:
     build: .
+    environment:
+      DB_NAME: govwifi_test
+      DB_PASS: root
+      DB_USER: root
+      DB_HOSTNAME: db
+      FRONTEND_API_KEY: supersecret
```
Comment (adrianclay) [inline docker-compose.yml]: Is this required?

File: docker-compose.yml [hunk 4]
```diff
@@ -1,7 +1,23 @@
...
+    links:
+      - db
     ports:
       - "8080:8080"
```
Comment (adrianclay) [inline docker-compose.yml]: Unrelated to this PR, but looks like we'll have some trouble in the near future with port conflicts locally.

File: app.rb [hunk 5]
```diff
@@ -1,7 +1,31 @@
...
+  end
+
+  get '/authorize/user/:user_name/*' do
+    authorize_user(params['user_name'])
+  end
+
+private
+
+  def authorize_user(user_name)
+    user = connect_to_db[:userdetails].select(Sequel[:password]).first(username: user_name)
```
Comment (adrianclay) [inline app.rb]: Maybe extract to a method.  On the fence about this line.

File: docker-compose.yml (hunk 4 above)
Comment (RyanMacG) [reply docker-compose.yml]: Cross that bridge when we come to it I guess

File: spec/app_spec.rb (hunk 2 above)
Comment (afoldesi-gds) [reply spec/app_spec.rb]: This is currently called from health checks from the ELB, with the exact username: HEALTH
We can possibly introduce a change to the RADIUS side to use the shortened version in every case, as at the authoriZation step the extra details are NOT used. (could also change the Z to S)

Comment (afoldesi-gds) [reply spec/app_spec.rb]: Um, I probably should clarify this. We don't currently use these parameters programmatically in the backend code.
However they contain the mac address and IP of the site which are parameters we can search for as a result of a BAU support request, to see if we received login attempts from a specific site at all, or from a device with a different username (eg. typo in the username)
Removing the long URLs for authorise, we would have to rely on post-auth only, leaving us with difficulties identifying for example an issue with the RADIUS server itself (auth received, but post-auth never sent).

Comment (adrianclay) [reply spec/app_spec.rb]: Would be good to document that in the README.md @RyanMacG

Comment (RyanMacG) [reply spec/app_spec.rb]: Have raised it as a trello ticket for a future PR

**Pull request merged**